src/V3ThreadPool.cpp

```diff
@@ -24,20 +24,24 @@
 constexpr unsigned int V3ThreadPool::FUTUREWAITFOR_MS;
 
 void V3ThreadPool::resize(unsigned n) VL_MT_UNSAFE VL_EXCLUDES(m_mutex)
-    VL_EXCLUDES(m_stoppedJobsMutex) {
+    VL_EXCLUDES(m_stoppedJobsMutex) VL_EXCLUDES(V3MtDisabledLock::instance()) {
+    // At least one thread (main)
+    n = std::max(1u, n);
+    if (n == (m_workers.size() + 1)) { return; }
     // This function is not thread-safe and can result in race between threads
     UASSERT(V3MutexConfig::s().lockConfig(),
             "Mutex config needs to be locked before starting ThreadPool");
     {
-        V3LockGuard lock{m_mutex};
         V3LockGuard stoppedJobsLock{m_stoppedJobsMutex};
+        V3LockGuard lock{m_mutex};
```

Comment:
You are acquiring two locks, be careful of lock inversion problems.  Audit all usages of both locks for that, and add comments near both mutexes saying that this needs to be checked when either mutex is used.

Reply:
I've audited all usages and they are correct. I've also added comment.